cpp/src/arrow/compute/kernels/cast.h

```diff
@@ -35,10 +35,23 @@ class DataType;
 namespace compute {
 
 struct ARROW_EXPORT CastOptions {
-  CastOptions() : allow_int_overflow(false), allow_time_truncate(false) {}
+  CastOptions()
+      : allow_int_overflow(false),
+        allow_time_truncate(false),
+        allow_float_truncate(true) {}
```

Comment:
This will be set to False by default via https://github.com/apache/arrow/pull/2504